examples/training/classification/imagenet/basic_training.py

```diff
@@ -182,49 +202,114 @@ def augment(img, label):
         **eval(FLAGS.model_kwargs),
     )
 
-
 """
-Next, we pick an optimizer. Here we use Adam with a constant learning rate.
-Note that learning rate will decrease over time due to the ReduceLROnPlateau callback.
+Optional LR schedule with cosine decay instead of ReduceLROnPlateau
+TODO: Replace with Core Keras LRWarmup when it's released. This is a temporary solution.
 """
 
 
-optimizer = optimizers.SGD(
-    learning_rate=INITIAL_LEARNING_RATE, momentum=0.9, global_clipnorm=10
+def lr_warmup_cosine_decay(
+    global_step, warmup_steps, hold=0, total_steps=0, start_lr=0.0, target_lr=1e-3
+):
+    # Cosine decay
+    # There is no tf.pi :(
+    learning_rate = (
```

Comment:
Can this part be done by directly calling tf.keras.optimizers.schedules.CosineDecay?
`https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/CosineDecay`

Reply:
Good call. I'll update it to do the direct call. Might very well update it to reflect Innat's implementation which has switchable decay schedules